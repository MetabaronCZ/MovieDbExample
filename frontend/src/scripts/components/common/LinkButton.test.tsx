import { act } from '@testing-library/react';
import { describe, expect, it, jest } from '@jest/globals';
import { fireEvent, getByRole } from '@testing-library/dom';

import { LinkButton } from './LinkButton';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/common/LinkButton', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(
      <LinkButton onClick={() => null}>CONTENT</LinkButton>,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const button = getByRole(container, 'button');
    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent('CONTENT');
  });

  it('should render basic props', async () => {
    const { container } = await waitForComponent(
      <LinkButton id="ID" className="CLASSNAME" onClick={() => null}>
        CONTENT
      </LinkButton>,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const button = getByRole(container, 'button');
    expect(button).toBeInTheDocument();

    expect(button).toHaveAttribute('id', 'ID');
    expect(button).toHaveClass('CLASSNAME');
    expect(button).toHaveTextContent('CONTENT');
  });

  it('should be clickable', async () => {
    const onClick = jest.fn();

    const { container } = await waitForComponent(
      <LinkButton onClick={onClick}>CONTENT</LinkButton>,
      { wrapper: TestComponentWrapper },
    );
    expect(onClick).not.toBeCalled();

    const button = getByRole(container, 'button');
    expect(button).toBeInTheDocument();

    // click on button
    act(() => {
      fireEvent.click(button);
    });
    expect(onClick).toBeCalledTimes(1);
  });
});
