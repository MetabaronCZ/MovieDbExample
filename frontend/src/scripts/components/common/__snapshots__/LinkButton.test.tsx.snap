// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/LinkButton should render basic props 1`] = `
.c0 {
  display: block;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  cursor: not-allowed;
}

.c1 {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  color: inherit;
  text-decoration: underline;
  display: inline-block;
}

.c1:hover {
  text-decoration: none;
}

.c1:focus {
  outline: 1px dashed #002d56;
  outline-color: inherit;
}

<div>
  <button
    class="c0 c1 CLASSNAME"
    id="ID"
    type="button"
  >
    CONTENT
  </button>
</div>
`;

exports[`components/common/LinkButton should render properly 1`] = `
.c0 {
  display: block;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  cursor: not-allowed;
}

.c1 {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  color: inherit;
  text-decoration: underline;
  display: inline-block;
}

.c1:hover {
  text-decoration: none;
}

.c1:focus {
  outline: 1px dashed #002d56;
  outline-color: inherit;
}

<div>
  <button
    class="c0 c1"
    type="button"
  >
    CONTENT
  </button>
</div>
`;
